sites/all/modules/reol_feed/reol_feed.module

```diff
@@ -344,3 +351,62 @@ function _reol_feed_br2nl($string) {
 function reol_feed_convert_html($html) {
   return strip_tags(_reol_feed_br2nl($html));
 }
+
+/**
+ * Provide a json feed for newsletter.
+ *
+ * @param $node_type
+ *   The type of node to display a feed of.
+ * @return array
+ *   An array of items to display in the feed.
+ */
+function _reol_feed_json_node($node_type) {
+  $supported_node_types = array(
+    'article'
+  );
+
+  if(!in_array($node_type, $supported_node_types)) {
+    return NULL;
+  }
+
+  $base_url = $GLOBALS['base_url'];
+
+  // Cache the page.
+  $newsletter_data = &drupal_static(__FUNCTION__);
+  if (!isset($newsletter_data)) {
+    if ($cache = cache_get('reol_feed_data')) {
+      $newsletter_data = $cache->data;
+    }
+    else {
+      // Fetch the relevant nodes.
+      $node_query = new EntityFieldQuery();
+      $node_query->entityCondition('entity_type', 'node');
+
+      if (array_key_exists($node_type, node_type_get_names()))  {
```

Comment:
If this is checked in the access callback - allowed node type - this is not required.